src/OrleansRuntime/Scheduler/TaskWorkItem.cs

```diff
@@ -17,13 +18,15 @@ internal class TaskWorkItem : WorkItemBase
         /// <param name="sched">Scheduler to execute this Task action. A value of null means use the Orleans system scheduler.</param>
         /// <param name="t">Task to be performed</param>
         /// <param name="context">Execution context</param>
-        internal TaskWorkItem(ITaskScheduler sched, Task t, ISchedulingContext context)
+        /// <param name="loggerFactory">logger factory used to create loggers</param>
+        internal TaskWorkItem(ITaskScheduler sched, Task t, ISchedulingContext context, ILoggerFactory loggerFactory)
         {
             scheduler = sched;
             task = t;
             SchedulingContext = context;
+            this.logger = loggerFactory.CreateLogger<TaskWorkItem>();
```

Comment:
TaskWorkItem and InvokeWorkItem need the same treatment as CallbackData and have a cached logger passed in, as they are also created VERY frequently (actually even more than the others you already addressed).
ActivationTaskScheduler is not created terribly often, but if it can get the same treatment without too many changes, do that one too.

Reply:
will do